### ISSUE TITLE
video playback stops when phone switches to or from battery saver

### ISSUE THREAD

**Issue by melanieLongden** [futo-org/grayjay-android#1453]
### What happened?

Fairphone 4, android 13
this issue has persisted accross multiple updates to plugin and app.

steps to recreate:
1.  Start watching a youtube video
2. Wait for battery saver to turn on based on %
3. see video playback cease

### Grayjay Version

264

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

209

### When do you experience the issue?

- [X] While logged in
- [ ] While logged out
- [ ] N/A

**Comment by support-grayjay** [comment]
I wasn't able to reproduce this on Android 15 with my Google Pixel 7 Pro.

**Comment by SdoggaMan** [comment]
FWIW, this doesn't happen on Moto Edge 30 Pro, Android 13 (I should know, given how often I slip into -10%!)

**Comment by nastynick64** [comment]
Specs: Pixel 5 running Android 14 on GrapheneOS

I have experienced the same issue. Basically whenever I'm watching a video, sorting a playlist, going through search list...etc, if the battery saver gets toggled (manually or automatically), it gets rid of that page and go back to home. It essentially restarts the app in-place. But, this isn't an issue exclusive to GrayJay. Same thing happens when using Brave browser, and Vanadium browser (GrapheneOSes built-in browser). I'll have bunch of tabs open and it will just restart the browser in-place. But one app that doesn't seem to suffer from this issue at all is VLC. And most simpler apps don't seem to have this issue (Messaging apps, document viewers, etc). I know the last couple things don't relate explicitly to GrayJay, but I just wanted mention it, just in case that can be helpful somehow.

**Comment by melanieLongden** [comment]
I've got a couple of screen captures of it incase this is useful. One is plugging into charge whilst on battery saver, the other is turning on battery saver
I don't know if this something unique to greyjay, however I will note I haven't noticed this effect on other apps.

It springs to mind that my phone changes from light to dark theme based on battery saver, it's possible that may have something to do with it. 

https://github.com/user-attachments/assets/19b2aca5-132f-4df9-9842-4fde9a0e1246


https://github.com/user-attachments/assets/529d6721-2e6f-419c-80bc-1e12f2b3d28e

**Comment by SdoggaMan** [comment]
Aha - i'm in dark mode at all times, so when I go into battery saver, I don't flip modes. That backs up why I don't see this issue!

Great find @melanieLongden .

**Comment by Splitpeach** [comment]
> Update - Just did a quick test and it with phone on dark mode on both sides of battery saver, the issue doesn't occur. So it seems that the phone flipping from light mode to dark mode or vice versa causes this issue

Are you using a custom ROM? I assume you are if on a Fairphone.

Im also having this issue. I'm on Calyx OS

**Comment by melanieLongden** [comment]
> > Update - Just did a quick test and it with phone on dark mode on both sides of battery saver, the issue doesn't occur. So it seems that the phone flipping from light mode to dark mode or vice versa causes this issue
> 
> Are you using a custom ROM? I assume you are if on a Fairphone.
> 
> Im also having this issue. I'm on Calyx OS

I don't believe I am, standard android. I do intend on picking myself a custom rom in future but right now I'm still weaning myself off from android

**Comment by Splitpeach** [comment]
This is becoming very irritating now. Why is this being ignored?

**Comment by melanieLongden** [comment]
@Splitpeach It's quite frustrating for me, a quick fix would be to simply switch your phone theme to be 'dark' both in and out of battery saver, i think that should fix the issue. 

I would probably promote the issue more but I am attempting to cut down my phone usage and I've made enough progress that I'm not so affected by this. But yes I do hope that that this issue will be fixed soon

**Comment by melanieLongden** [comment]
Wonderful! Thank you : )

**Comment by Zvonimir-FUTO** [comment]
Fixed in v312.
